### ISSUE TITLE
Some more stuff for machine learning

### ISSUE THREAD

**Issue by currymj** [Gabriel439/post-rfc#98]
I think there are some additional promising libraries in the area of machine learning.

- Most exciting from a propaganda perspective, there are Haskell bindings for Tensorflow. The library is hosted by the tensorflow organization but states that it is unofficial. It seems like it might be someone's 20% project.
- The `ad` library for automatic differentiation is arguably best-in-class. Computing derivatives/gradients is a fundamental operation for a lot of machine learning. 
- There are several still-immature pure Haskell neural network libraries that use `ad` for backprop, including `neural` and `grenade`. The type system is used to ensure that input/output types of each layer of the network are consistent.
- The `monad-bayes` library is under rapid development (new commits ~daily) and has the potential to be very good for probabilistic programming. It is only on github now, won't be published to hackage until it is more stable.

I think `ad` could probably be considered mature. The rest less so.

**Comment by Gabriella439** [comment]
I added `tensorflow`, `ad` and `grenade` to the machine learning section in 3ad4a723caf3d14bc915b414e2893724be1cbde5

I left out `neural` and `monad-bayes` for now since they seemed to be lower quality than `grenade`, but I'll keep this issue open to remind myself to revisit them later on
